### PR TITLE
[DOCS-3698] Update docs for custom instrumentation using the latest api v0

### PR DESCRIPTION
**Pull request by labbati**
### What does this PR do?

Provides examples and samples about what is the recommended way to do custom instrumentation. Moreover it updates some snippets to remove reference to our legacy api.

### Motivation

Some of our snippets were using an API that we have now long deprecated. Moreover we had many bit here and there but not a consistent example with multiple interesting use cases.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by kayayarai** [comment]
opened https://datadoghq.atlassian.net/browse/DOCS-3698 to track edit

**Review by kayayarai — APPROVED** [review]
Gave this a pretty thorough review, @labbati , and changed it a bit. Let me know if I messed anything up. If you think it's good, I'll merge it.

File: content/en/tracing/setup_overview/custom_instrumentation/php.md [hunk 1]
```diff
@@ -22,42 +22,232 @@ further_reading:
...
+        $span->meta['app.cache.item_id'] = $ret; // The returned value
+    }
+);
+   {{< /code-block >}}
+
+   <div class="alert alert-warning">
+   When you set tags, to avoid overwriting existing tags automatically added by the Datadog core instrumentation, <strong>do write <code>$span->meta['mytag'] = 'value'</code></strong>. Do not write <code>$span->meta = ['mytag' => 'value']</code>.
+   </div>
+
+6. In the sample code, `SampleRegistry::faultyMethod` generates an exception. There is nothing your have to do with regards to custom instrumentation. If the method is instrumented, the default exception reporting mechanism takes care of attaching the exception message and the stack trace.
```
Comment (labbati) [inline content/en/tracing/setup_overview/custom_instrumentation/php.md:165]:
```suggestion
6. In the sample code, `SampleRegistry::faultyMethod` generates an exception. There is nothing you have to do with regards to custom instrumentation. If the method is instrumented, the default exception reporting mechanism takes care of attaching the exception message and the stack trace.
```

**Review by labbati — COMMENTED** [review]
wow @kayayarai the page after your edits looks awesome and much more professional. Thank you for that. I only notices a little left over from my terrible english, then from my point of view we can merge.

Thank you again

**Pull request merged**